RC 0.0.6

jklenzing opened the pull request:
# Description


# Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

via pytest and Github Actions config

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json
- [x] doc tests pass locally
- [x] activate this branch on [readthedocs](https://pysatnasa.readthedocs.io/en/rc_0_0_6/)
- [x] Upload the test distribution to the testPyPi [server](https://test.pypi.org/project/pysatNASA/0.0.6rc1/)
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages)
- [x] Create a wiki page with the release text (put link here)
- [x] After approval, confirm all changes from the RC pull are listed in the wiki release text.- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

aburrell committed:
BUG: fixed init metadata

Fixed init metadata by:
- moving all imports to top, and
- robustly importing metadata submodule.

aburrell committed:
DOC: updated changelog

Added the changes to the log.

jklenzing committed:
Merge pull request #206 from pysat/inst/reach

ENH: REACH data

jklenzing committed:
Merge branch 'develop' into maven_mgs

aburrell committed:
Merge pull request #210 from pysat/improved_init

Improved init

jklenzing committed:
Merge pull request #200 from pysat/maven_mgs

Maven mgs

jklenzing committed:
Merge branch 'develop' into enh/103_104_zip

jklenzing committed:
DOC: update changelog

jklenzing committed:
BUG: always return empty objects if no data

jklenzing committed:
DOC: update changelog

jklenzing committed:
BUG: update test

jklenzing committed:
ENH: add test data

jklenzing committed:
STY: use year for dir

jklenzing committed:
Remove test data

jklenzing committed:
STY: remove gzip

jklenzing committed:
DOC: update changelog

jklenzing committed:
DOC: add comment

jklenzing committed:
TST: add warning test

jklenzing committed:
STY: pep8

jklenzing committed:
Merge pull request #209 from pysat/enh/103_104_zip

ENH: unzip support for download

jklenzing committed:
MAINT: update clean routine

jklenzing committed:
DOC: update changelog

jklenzing committed:
MAINT: allow old clean routine if needed

jklenzing committed:
STY: improve style

jklenzing committed:
MAINT: github token

jklenzing committed:
MAINT: github run id

jklenzing committed:
BUG: github job

jklenzing committed:
BUG: new de2 load

jklenzing committed:
Merge pull request #214 from pysat/maint/cnofs_vefi

MAINT: cnofs vefi magnetometer data

jklenzing committed:
Update CHANGELOG.md

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>

jklenzing committed:
Merge branch 'develop' into bug/cdf_load

jklenzing committed:
BUG: coveralls

jklenzing committed:
MAINT: remove coveralls cap

jklenzing committed:
TST: use coveralls app

jklenzing committed:
MAINT: add finished to coveralls

jklenzing committed:
MAINT: use command line coveralls

jklenzing committed:
BUG: adjust env

jklenzing committed:
STY: update workflow syntax

jklenzing committed:
BUG: name

jklenzing committed:
BUG: empty meta in xarray

jklenzing committed:
TST: expand test options

jklenzing committed:
TST: add test for bad files

jklenzing committed:
Merge pull request #212 from pysat/bug/cdf_load

BUG: cdf load

jklenzing committed:
Merge branch 'develop' into bug/de2_vefi

jklenzing committed:
BUG: drop use_cdflib

jklenzing committed:
BUG: version cap for sphinx rtd

jklenzing committed:
BUG: pop not drop

jklenzing committed:
DOC: add saber ref

jklenzing committed:
DOC: update changelog

jklenzing committed:
MAINT: sphinx rtd theme cap

jklenzing committed:
DOC: add saber ref

jklenzing committed:
DOC: update changelog

jklenzing committed:
BUG: drop extra dimension for de2_vefi

jklenzing committed:
DEP: deprecate support for '' tag for vefi

jklenzing committed:
ENH: add vefimagb instrument

jklenzing committed:
DOC: update changelog

jklenzing committed:
Merge pull request #216 from pysat/doc/saber

DOC: saber

jklenzing committed:
Merge branch 'develop' into bug/de2_vefi

jklenzing committed:
Apply suggestions from code review

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>

jklenzing committed:
STY: remove kwargs

jklenzing committed:
BUG: add additional skip check

jklenzing committed:
BUG: update tests

jklenzing committed:
DOC: add comments

jklenzing committed:
BUG: set de2 vefimagb to clean warn

jklenzing committed:
BUG: kwargs

jklenzing committed:
Merge branch 'develop' into guvi_imaging_fix

jklenzing committed:
MAINT: update rc tests

jklenzing committed:
Merge pull request #217 from pysat/bug/de2_vefi

BUG: de2 vefi load

jklenzing committed:
Merge pull request #219 from pysat/maint/rc_tests

MAINT: update RC tests

jklenzing committed:
Update pysatNASA/tests/test_instruments.py

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>

jklenzing committed:
Merge pull request #220 from pysat/develop

Pull Develop changes into guvi branch

jklenzing committed:
BUG: cdaweb download

jklenzing committed:
MAINT: tests for icon ivm b

jklenzing committed:
DOC: update changelog

jklenzing committed:
STY: use new test flags

jklenzing committed:
STY: pep8

jklenzing committed:
STY: turn off verbose on pytest

jklenzing committed:
MAINT: update new tests flag

jklenzing committed:
STY: PEP8

jklenzing committed:
BUG: fix rc action

aburrell committed:
ENH: add SDR disk tags to DMSP SSUSI

Update the sub-module attributes and put a placeholder for the load method.

aburrell committed:
BUG: fix JHUAPL time

Fix time where microseconds can become too big from rounding.

aburrell committed:
STY: removed type equality

Updated type equalities to use `isinstance`.

aburrell committed:
TST: added an ignore list for links

Added a `linkcheck_ignore` statement, as the GATS page checks to see if the link request is human or machine.

aburrell committed:
BUG: fixed inners assignment

Ensured that only dimensions with data are renamed.  Also updated test date to highlight a time with better data coverage across all instruments.

aburrell committed:
TST: update GUVI test dates

Update TIMED GUVI test dates, as coverage needs are different for different instruments.

jklenzing committed:
Update pysatNASA/instruments/methods/cdaweb.py

jklenzing committed:
DOC: move link to comments

jklenzing committed:
Merge branch 'bug/cdaweb_clean' of https://github.com/pysat/pysatNASA into bug/cdaweb_clean

jklenzing committed:
REV: move link back

jklenzing committed:
Merge pull request #223 from pysat/bug/cdaweb_clean

BUG/MAINT: cdaweb downloads, ICON IVM-B tests, new flag to skip new tests

jklenzing committed:
Merge branch 'develop' into guvi_imaging_fix

jklenzing committed:
MAINT: remove new test flag from timed_guvi

jklenzing committed:
Update pysatNASA/instruments/timed_guvi.py

jklenzing committed:
MAINT: only skip high res data

aburrell committed:
Merge pull request #198 from pysat/guvi_imaging_fix

GUVI imaging updates

aburrell committed:
Merge pull request #224 from pysat/develop

Develop update for new working branch

aburrell committed:
DOC: updated index depth

Updated the index depth to make the front page more legible.

aburrell committed:
DOC: updated changelog

Added new instrument and updated release year to changelog.

aburrell committed:
ENH: updated jhuapl methods

Updated the JHU APL methods by:
- generalizing the `load_sdr_aurora` function to handle SSUSI data, and
- added a generalized cleaning function for data with the DQI flag.

aburrell committed:
ENH: updated SSUSI clean and load

Updated the SSUSI clean and load methods to handle SDR disk images.

aburrell committed:
ENH: updated GUVI cleaning

Updated the GUVI cleaning to:
- fix bugs that prevented any actual cleaning,
- use a generalized function, and
- group the dusty/dirty flags together.

aburrell committed:
STY: replaced tabs

Replaced tabs with spaces.

aburrell committed:
BUG: fixed inputs

Removed input that was added to the wrong load function.

aburrell committed:
BUG: fixed errors in DMSP SSUSI

Fixed missing import, wrong tag name, and bad input argument found in the DMSP SSUSI Instrument.

aburrell committed:
TST: created unit tests for JHUAPL methods

Created unit tests for the non-data loading functions in the JHUAPL methods.

aburrell committed:
BUG: fixed test bugs

Fix test bugs for identifying NaN fill values and the input for the epoch time calculation.

aburrell committed:
STY: fixed hanging indent

Fixed hanging indent and removed line breaks with `\`.

aburrell committed:
BUG: fixed if condition

Fixed if condition to allow SSUSI channel names to be specified.

aburrell committed:
BUG: fixed `nCross` identification

Fixed the way cross-track variables are identified, as this varies from GUVI to SSUSI.

aburrell committed:
BUG: normalize auroral time name

As no auroral data has been found labelled 'night' yet, use the GUVI standards.

aburrell committed:
STY: fixed hanging indent

Fixed a bad hanging indent.

aburrell committed:
Merge pull request #225 from pysat/dmsp_ssusi_sdr_disk

ENH: DMSP SSUSI SDR disk images

aburrell committed:
DOC: added acknowledgements

Added acknowledgements documentation to the repository and documentation.

aburrell committed:
DOC: added pub release to docs

Added the NRL pub release statement to the documentation.

aburrell committed:
DOC: top level pub release

Added the NRL pub release statement to the top level code files.

aburrell committed:
DOC: constellation pub release

Added the NRL pub release statement to the constellation files.

aburrell committed:
DOC: updated instrument with pub release

Added the NRL pub release to the instrument sub-module files.

aburrell committed:
DOC: updated instrument methods headers

Added NRL pub release to the instrument method sub-modules.

aburrell committed:
DOC: updated CDAWeb template

Updated the CDAWeb instrument template and docstring to include the NRL pub release and instruct users to not include it in new files.

aburrell committed:
DOC: updated test headers

Added NRL pub release to the pysatNASA unit and integration tests.

aburrell committed:
MAINT: updated .gitignore

Added the current default build directory for local documentation to .gitignore.

aburrell committed:
DOC: updated linked text

Updated text to:
- have the correct line length,
- use better grammar, and
- not have duplicate text with different links.

aburrell committed:
MAINT: updated conf.py

Updated the interspinx mapping format as suggested in the deprecation warning.

aburrell committed:
BUG: fixed typo

Fixed a typo in the index file.

aburrell committed:
BUG: fixed supported Instrument formatting

Fixed supported instrument reference links, automodule calls, and title underlines.

jklenzing committed:
DOC: udpate zenodo

jklenzing committed:
MAINT: NEP29 apr 2024

jklenzing committed:
DOC: update supported versions

jklenzing committed:
DOC: update changelog

jklenzing committed:
BUG: no new tests for ssusi

jklenzing committed:
BUG: comment out buggy tests

jklenzing committed:
BUG: turn off new tests for ICON FUV

jklenzing committed:
BUG: remove icon clean tests

jklenzing committed:
BUG: manual install of git

jklenzing committed:
BUG: git branch

jklenzing committed:
BUG: spelling

jklenzing committed:
BUG: use python setup

jklenzing committed:
BUG: hdf

jklenzing committed:
BUG: develop

aburrell committed:
STY: generalized `concat_data` method

Extracted the general portion of the GUVI `concat_data` method, to allow use in the DMSP SSUSI data.

aburrell committed:
STY: use JHU APL concat

Use the new JHU APL concat function in local method.

aburrell committed:
BUG: add custom `concat_data` method

Use the JHU APL `concat_data` function in this Instrument.

aburrell committed:
DOC: udpated changelog

Modified changelog to reflect changes in this pull request.

aburrell committed:
BUG: update DMSP SSUSI test flags

Updated the DMSP SSUSI test flags to only test for warnings in the clean routine when appropriate and to skip the new tests for the high resolution imaging data.

aburrell committed:
TST: expand test skipping

Tests still failing due to too much memory, add another skip.

aburrell committed:
Merge branch 'maint/nep29_apr24' into ssusi_bug

jklenzing committed:
Update main.yml

jklenzing committed:
Update main.yml

jklenzing committed:
BUG: always run finish

jklenzing committed:
BUG: GA syntax

jklenzing committed:
BUG: turn off new tests for maven sep

jklenzing committed:
Try coveralls app

aburrell committed:
BUG: switch to coveralls app

Switch to coveralls app from coveralls action.

aburrell committed:
BUG: fix yaml syntax

Attempt to fix yaml syntax.

aburrell committed:
BUG: more yaml fixes

Another attempt to fix yaml syntax.

aburrell committed:
BUG: re-added steps

Re-added `steps` to yaml.

aburrell committed:
BUG: added if to yaml

Added a missing if statement to the yaml and fixed some of the step names.

aburrell committed:
BUG: fixed secret name

Fixed the name of the GitHub secret.

aburrell committed:
BUG: specify coverage file

Specify the coverage file in the macOS call.

aburrell committed:
BUG: added debug line

Coveralls app isn't working on mac, no idea why.  Checking that coverage file exists.

aburrell committed:
BUG: use coveralls repo token

Use the Coveralls Repository Token, which I just added to the GitHub repository.

aburrell committed:
BUG: specify format

Try specifying the format for the MacOS coverage call.

aburrell committed:
BUG: remove "extra" options

Try removing as many options as possible.

aburrell committed:
Revert "BUG: remove "extra" options"

This reverts commit e35b977f573bc18642fe25afb8396dc1a96c1029.

aburrell committed:
BUG: add verbose mode

Put coveralls in verbose mode in preparation for help call.

aburrell committed:
BUG: specify xml output

Specify coverage xml output.

aburrell committed:
BUG: remove tabs from yaml

Remove the tabs from the yaml.

aburrell committed:
BUG: update xml

Fix the name of the xml to reflect reality instead of the docs.

aburrell committed:
BUG: made coveralls update

Made changes suggested by James from Coveralls.

aburrell committed:
BUG: added debug

Added a debug statement, as suggested for better feedback.

aburrell committed:
BUG: fixed large hours

Fixed a bug with roll-over hours.

jklenzing committed:
NEP29 for June 2024

jklenzing committed:
Merge pull request #227 from pysat/maint/nep29_apr24

MAINT: NEP29 Apr 2024, zenodo updates

jklenzing committed:
Merge branch 'develop' into nrl_pub_release

jklenzing committed:
Merge branch 'develop' into ssusi_bug

aburrell committed:
Merge pull request #231 from pysat/ssusi_bug

SSUSI bug

aburrell committed:
DOC: fixed acknowledgements

Fixed the headers in the acknowledgements document.

Co-authored-by: Jeff Klenzing <19592220+jklenzing@users.noreply.github.com>

aburrell committed:
DOC: Update ACKNOWLEDGEMENTS.md

Fixed length of section underline.

aburrell committed:
Merge pull request #228 from pysat/nrl_pub_release

NRL pub release

jklenzing committed:
MAINT: update version

jklenzing committed:
MAINT: add tentative release date

jklenzing committed:
MAINT: add new pytest marker

jklenzing committed:
MAINT: logger.warning

jklenzing committed:
MAINT: pandas freq

jklenzing committed:
MAINT: ffill and bfill

jklenzing committed:
MAINT: iloc

jklenzing committed:
MAINT: pysat headers

jklenzing committed:
DOC: update changelog

jklenzing committed:
MAINT: update GA versions

jklenzing committed:
DOC: update changelog

jklenzing committed:
MAINT: usage of dims

jklenzing committed:
Merge pull request #235 from pysat/maint/future_warn_2024

MAINT: Future Warnings for numpy, pandas, pysat

jklenzing commented:
RC1 is live on test pypi. https://test.pypi.org/project/pysatNASA/0.0.6rc1/

aburrell commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Module for the MAVEN insitu instruments.
```
```suggestion
"""Module for the MAVEN in situ instruments.
```

aburrell commented on pysatNASA/instruments/maven_mag.py:
```diff
@@ -0,0 +1,101 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Module for the MAVEN mag instrument.
```
```suggestion
"""Module for the MAVEN magnetometer.
```

aburrell commented on pysatNASA/instruments/maven_sep.py:
```diff
@@ -0,0 +1,116 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Module for the MAVEN sep instrument.
```
```suggestion
"""Module for the MAVEN Solar Energetic Particle instrument.
```

aburrell commented on pysatNASA/instruments/maven_sep.py:
```diff
@@ -0,0 +1,116 @@
+::
+    import pysat
+
+    insitu = pysat.Instrument(platform='maven', name='sep', inst_id='s1')
+    insitu.download(dt.datetime(2020, 1, 1), dt.datetime(2020, 1, 31))
+    insitu.load(2020, 1, use_header=True)
```
```suggestion
    sep = pysat.Instrument(platform='maven', name='sep', inst_id='s1')
    sep.download(dt.datetime(2020, 1, 1), dt.datetime(2020, 1, 31))
    sep.load(2020, 1, use_header=True)
```

aburrell commented on pysatNASA/instruments/maven_sep.py:
```diff
@@ -0,0 +1,116 @@
+
+platform = 'maven'
+name = 'sep'
+tags = {'': 'Level 2 Solar Energetic Particle data'}
+inst_ids = {'s1': [''], 's2': ['']}
```
This would be better as 'l2' for the tag, as a future-proofing effort.

rstoneback replied:
I think most instruments have used l2 (level-2) public science data as a default. 
Does l2 mean something different here than say C/NOFS or ICON?

jklenzing replied:
It means extra typing for end-users, and there is no plan to currently bring other levels in.

aburrell commented on pysatNASA/instruments/methods/maven.py:
```diff
@@ -0,0 +1,45 @@
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Provides non-instrument specific routines for MAVEN data.
+
+Created on Thu Jul 13 11:21:01 2023
+
+@author: tesman
```
These bits are not normally included in the docstring.

aburrell commented on pysatNASA/instruments/reach_dosimeter.py:
```diff
@@ -0,0 +1,172 @@
+    '101', '102', '105', '108', '113', '114', '115', '116', '133', '134', '135',
+    '136', '137', '138', '139', '140', '148', '149', '162', '163', '164', '165',
+    '166', '169', '170', '171', '172', '173', '175', '176', '180', '181'
+
+
```
```suggestion

```

aburrell commented on pysatNASA/instruments/reach_dosimeter.py:
```diff
@@ -0,0 +1,172 @@
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+
+
```
```suggestion

```

aburrell commented on pysatNASA/instruments/reach_dosimeter.py:
```diff
@@ -0,0 +1,172 @@
+    self.acknowledgements = self.meta.header.Acknowledgement
+
+    return
+
+
```
```suggestion

```

aburrell commented on pysatNASA/instruments/reach_dosimeter.py:
```diff
@@ -0,0 +1,172 @@
+    meta.header = MetaHeader(new_header)
+
+    return data, meta
+
+
```
```suggestion

```

aburrell replied:
I think it's just one enter line after a function?

aburrell commented on pyproject.toml:
```diff
@@ -46,13 +46,14 @@ dependencies = [
   "pandas",
   "pysat >= 3.1",
```
```suggestion
  "pysat >= 3.2",
```

Because of the header flags, no?

rstoneback commented on docs/supported_instruments.rst:
```diff
@@ -97,6 +97,14 @@ DE2 VEFI
 
+.. _de2_vefimag:
+
+DE2 VEFIMAGB
```
Is the magnetometer part of the VEFI package? Seems like putting this dataset under the 'mag' tag is more in line with the organizational structure, including the C/NOFS VEFI support, rather than as its own instrument.

rstoneback commented on pysatNASA/instruments/de2_vefi.py:
```diff
@@ -89,10 +114,61 @@
+    Returns
+    -------
+    data : pds.DataFrame
+        A pandas DataFrame with data prepared for the `pysat.Instrument`.
```
Update to note support for both pandas and xarray as well as what conditions each format is returned.

jklenzing replied:
This only returns pandas as output. xarray is used to convert the data for the blank tag, which is deprecated.

rstoneback commented on pysatNASA/instruments/de2_vefi.py:
```diff
@@ -89,10 +114,61 @@
+    """
+
+    if tag == '':
+        # Warn user that e-field data is dropped.
```
Is there docstring documentation describing the dropping e-field thing?

jklenzing replied:
updated in the warnings

rstoneback commented on pysatNASA/instruments/de2_vefimagb.py:
```diff
@@ -0,0 +1,177 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Module for the DE2 VEFI instrument.
```
Repeated question, more directly placed. Why are there two DE2 VEFI instrument modules?

jklenzing replied:
See issue and pull referenced above.

rstoneback commented on pysatNASA/instruments/de2_vefimagb.py:
```diff
@@ -0,0 +1,177 @@
+
+# Set the load routine
+def load(fnames, tag='', inst_id='', **kwargs):
+    """Load DE2 VEFI data.
```
Tag support indicates both electric and magnetic field data is loaded. Comment depends upon if this independent instrument file continues after review.

jklenzing replied:
This instrument is required.

rstoneback commented on pysatNASA/instruments/dmsp_ssusi.py:
```diff
@@ -79,16 +92,74 @@
+
+
+def concat_data(self, new_data, combine_times=False, **kwargs):
+    """Concatonate data to self.data for DMSP SSUSI data.
```
```suggestion
    """Concatenate data to self.data for DMSP SSUSI data.
```

rstoneback commented on pysatNASA/instruments/dmsp_ssusi.py:
```diff
@@ -79,16 +92,74 @@
+    Parameters
+    ----------
+    new_data : xarray.Dataset or list of such objects
+        New data objects to be concatonated
```
```suggestion
        New data objects to be concatenated
```

rstoneback commented on pysatNASA/instruments/dmsp_ssusi.py:
```diff
@@ -79,16 +92,74 @@
+    if self.tag in ['sdr-disk', 'sdr2-dist']:
+        time_dims.append('time_auroral')
+
+    # Concatonate using the appropriate method for the number of time
```
```suggestion
    # Concatenate using the appropriate method for the number of time
```

rstoneback commented on pysatNASA/instruments/dmsp_ssusi.py:
```diff
@@ -52,6 +62,7 @@
 import datetime as dt
 import functools
 
+import pysat
```
```suggestion
from pysat import logger
```
(?) I remember there was some kind of weirdness with importing and the logger in particular. At this point though I don't remember what it is :( Something that came up during testing previously. I do remember it occurred with clearing a pysat import and reimporting the package though. I acknowledge it makes much of this comment less than perfectly helpful.Hopefully by writing it down it will free the memory before final merge.

rstoneback replied:
Upon further review, I now actually think the from pysat import logger is the problem. I'm keeping the back and forth on my thoughts as documentation for any future browsers that writing things down is essential.

Plus logging has been moved from some instrument files into general files. A heads up to those that may now about the existence of from pysat import logger in other areas.

rstoneback commented on pysatNASA/instruments/formosat1_ivm.py:
```diff
@@ -1,4 +1,12 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
 """Module for the Formosat-1 IVM instrument.
```
From a user perspective, unfortunate side-effect of funding patterns that formosat-x satellite data is spread across multiple packages. I don't have a solution in this comment box but it would be easier on users if there was but one place, or specific user directions. Perhaps a telecom topic.

jklenzing replied:
Add it to the agenda

rstoneback commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+
+platform = 'maven'
+name = 'insitu_kp'
+tags = {'': 'in situ Key Parameter data'}
```
mixed case

aburrell replied:
I believe "Key Parameter" is a proper name, and so allowed to be capitalized.  @jklenzing @t-esman please correct me if I'm wrong.

rstoneback commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+# ----------------------------------------------------------------------------
+"""Module for the MAVEN insitu instruments.
+
+Supports the in situ Key Parameter (kp) data from multiple instruments
```
I had thought this kp was like the Kp for the Earth, the multi-hour planetary index related to magnetic field stuff. Unfortunate bit of overlap. 

Double checking on the case for kp, it isn't KP? Also, what is a key parameter for Mars?

Is there a remote kp?

jklenzing replied:
Good catch, this should be KP. It is a Key Parameter (team's definition), which includes thermal plasma measurements, etc. Basically the measurements from all instruments that the science team is likely to use. I don't think it includes all parameters. Should be more info in the refs.

rstoneback replied:
Double-checking on the update of (kp) to (KP)

jklenzing replied:
This should be KP and currently is when I look at the up to date files

rstoneback commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+::
+    import pysat
+
+    insitu = pysat.Instrument(platform='maven', name='insitu_kp')
```
```suggestion
    insitukp = pysat.Instrument(platform='maven', name='insitu_kp')
```

or

 ```suggestion
    kp = pysat.Instrument(platform='maven', name='insitu_kp')
```
or
```suggestion
    insitukp = pysat.Instrument(platform='maven', name='insitu_kp')
```


for general consistency with a later comment on a different instrument that also used insitu.

rstoneback commented on pysatNASA/instruments/methods/_cdf.py:
```diff
@@ -437,13 +445,14 @@ def to_pysat(self, flatten_twod=True,
         for varname, df in cdata.items():
             if varname not in ('Epoch', 'DATE'):
-                if type(df) == pds.Series:
+                if isinstance(df, pds.Series):
```
I get that Python has stated they don't like type checks because code should use duck-typing. If it walks like a duck, etc. I also know that the isinstance is recommended over type for when type checks are put in. I don't get it though. This doesn't have to be answered here but if someone does know I'd love to hear it.

jklenzing replied:
Pretty sure I implemented this from either a FutureWarning or a linting suggestion.

jklenzing replied:
This is the specific error I get when I run flake8: 
```
E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
```

https://www.flake8rules.com/rules/E721.html

rstoneback commented on pysatNASA/instruments/methods/cdaweb.py:
```diff
@@ -566,6 +590,52 @@ def download(date_array, data_path, tag='', inst_id='', supported_tags=None,
+        Path to pysat archival directory.
+    fname : str
+        Name of file on the remote server.
+    temp_path : str
```
What happens is None is provided?

jklenzing replied:
A TypeError. D:

aburrell replied:
If that's the case, shouldn't this be an arg?  If not, there should be a more informative error.

jklenzing replied:
This whole function should probably be replaced with `_get_file` as it is not intended to be directly invoked by a user

rstoneback commented on pysatNASA/instruments/methods/cdaweb.py:
```diff
@@ -902,9 +972,12 @@ def cdas_list_remote_files(tag='', inst_id='', start=None, stop=None,
+    stop = pysat.utils.time.filter_datetime_input(stop)
+
+    # cdasws needs a time for the stop date.
+    stop += dt.timedelta(seconds=86399)
```
What happens with data that has less than a 1 second cadence? I also don't quite get how this interacts with the previous definitions of stop.

jklenzing replied:
cdasws uses stop differently from pysat. See https://github.com/pysat/pysatNASA/pull/223

rstoneback commented on pysatNASA/instruments/methods/iss.py:
```diff
@@ -0,0 +1,37 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Provides non-instrument specific routines for ISS data."""
```
```suggestion
"""Provides non-instrument specific routines for International Space Station (ISS) data."""
```

jklenzing replied:
We haven't spelled out acronyms elsewhere in methods, also is too many characters. Add to agenda.

rstoneback commented on pysatNASA/instruments/methods/omni.py:
```diff
@@ -32,15 +51,15 @@ def time_shift_to_magnetic_poles(inst):
     inst['Vx'] = inst['Vx'].interpolate('nearest')
-    inst['Vx'] = inst['Vx'].fillna(method='backfill')
-    inst['Vx'] = inst['Vx'].fillna(method='pad')
+    inst['Vx'] = inst['Vx'].bfill()
```
pandas updates?

jklenzing replied:
FutureWarning from pandas updates. The method kwargs are deprecated, new functions for backfill (bfill) and forward fill (ffill)

rstoneback commented on pysatNASA/instruments/reach_dosimeter.py:
```diff
@@ -0,0 +1,172 @@
+
+The Responsive Environmental Assessment Commercially Hosted (REACH)
+constellation is collection of 32 small sensors hosted on six orbital planes of
+the Iridium-Next space vehicles in low earth orbit. Each sensor contains two
```
```suggestion
the Iridium-Next space vehicles in low Earth orbit. Each sensor contains two
```

rstoneback commented on pysatNASA/instruments/templates/template_cdaweb_instrument.py:
```diff
@@ -1,8 +1,18 @@
 
 Copy and modify this file as needed when adding a new Instrument to pysat.
 
+DO NOT include the NRL distribution statement in your new file.
```
This seems weird and unexpected given I see the NRL statement across what seems to be all the files. I think a brief explanation would be helpful for future users and developers.

jklenzing replied:
```suggestion
DO NOT include the NRL distribution statement in your new file. This will have to be
approved by NRL separately before the versioned release.
```
@aburrell, does this work?

rstoneback commented on pysatNASA/instruments/timed_guvi.py:
```diff
@@ -86,20 +95,81 @@
+
+
+def concat_data(self, new_data, combine_times=False, **kwargs):
+    """Concatonate data to self.data for TIMED GUVI data.
```
```suggestion
    """Concatenate data to self.data for TIMED GUVI data.
```

rstoneback commented on pysatNASA/instruments/timed_guvi.py:
```diff
@@ -86,20 +95,81 @@
+    Parameters
+    ----------
+    new_data : xarray.Dataset or list of such objects
+        New data objects to be concatonated
```
```suggestion
        New data objects to be concatenated
```

rstoneback commented on pysatNASA/instruments/timed_guvi.py:
```diff
@@ -86,20 +95,81 @@
+    elif self.tag == 'sdr-spectrograph':
+        time_dims.extend(['time_gaim_day', 'time_gaim_night'])
+
+    # Concatonate using the appropriate method for the number of time
```
```suggestion
    # Concatenate using the appropriate method for the number of time
```

rstoneback commented on pysatNASA/tests/test_instruments.py:
```diff
@@ -34,13 +42,18 @@
 instruments['cdf'] = []
+
+# Create list of pandas instruments where this is not needed
+skip_cdf_list = ['de2_vefimagb']
```
For future users and maintainability, is there documentation somewhere about this?

jklenzing replied:
I'm pretty sure there's an issue about this at pysatCDF. :)

Technically we don't run any tests with pysatCDF anymore. See https://github.com/pysat/pysatNASA/issues/167

rstoneback replied:
Lame but I get it.

Aadarsh-Govada committed:
Try coveralls pip installation

Aadarsh-Govada committed:
Use verbose and showlocals pytest versions

Aadarsh-Govada committed:
Cap cdasws version at 1.8.2

Aadarsh-Govada committed:
Restore MacOS brew coveralls installation

Aadarsh-Govada committed:
Remove --quiet from coveralls installation on MacOS

Aadarsh-Govada committed:
Revert to rcfile coveralls publishing for MacOS

Aadarsh-Govada committed:
DOC: Update zenodo.json

Aadarsh-Govada committed:
Update CHANGELOG.md

jklenzing commented:
TODO:
- [ ] merge in other PRs
- [ ] Wait for SPDF to update metadata / fix bug (#236)

Aadarsh-Govada committed:
MAINT: Remove cdasws version cap

aburrell committed:
BUG: test returned output

Test returned output to provide an empty list if no remote files were found.

aburrell committed:
DOC: update changelog

Added a line to the changelog.

jklenzing committed:
MAINT: fix broken link

jklenzing committed:
DOC: update changelog

aburrell committed:
Merge pull request #238 from pysat/no_remote_files

Empty return without remote files

jklenzing committed:
Merge branch 'rc_0_0_6' into cover

jklenzing commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+# ----------------------------------------------------------------------------
+"""Module for the MAVEN insitu instruments.
+
+Supports the in situ Key Parameter (kp) data from multiple instruments
```
```suggestion
Supports the in situ Key Parameter (KP) data from multiple instruments
```

jklenzing commented on pysatNASA/instruments/methods/maven.py:
```diff
@@ -0,0 +1,45 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Provides non-instrument specific routines for MAVEN data.
+
+Created on Thu Jul 13 11:21:01 2023
+
+@author: tesman
+"""
```
```suggestion
"""Provides non-instrument specific routines for MAVEN data."""
```

jklenzing commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+    import pysat
+
+    insitu = pysat.Instrument(platform='maven', name='insitu_kp')
+    insitu.download(dt.datetime(2020, 1, 1), dt.datetime(2020, 1, 31))
+    insitu.load(2020, 1, use_header=True)
```
```suggestion
    insitu.download(dt.datetime(2020, 1, 1), dt.datetime(2020, 1, 31))
    insitu.load(2020, 1, use_header=True)
```
```suggestion
    insitukp.download(dt.datetime(2020, 1, 1), dt.datetime(2020, 1, 31))
    insitukp.load(2020, 1, use_header=True)
```

jklenzing committed:
Apply suggestions from code review

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>
Co-authored-by: Russell Stoneback <github@stoneris.com>

aburrell committed:
DOC: updated GOLD docstring

Updated the format in a GOLD docstring.

aburrell committed:
ENH: added SSUSI file format function

Added a function to provide the correct filename format for DMSP SSUSI data based on the date.

aburrell committed:
BUG: fix file format

Added custom `download`, `list_files`, and `list_remote_files` methods that account for the changes in the file format.

aburrell committed:
DOC: update changelog

Update the changelog to include a line describing the addition to the DMSP instrument methods.

aburrell committed:
TST: update instrument tests

Update the instrument tests.  If `supported_tags` is not present, do not require it.  It is only required if the Instrument has 'cdf' files.

aburrell committed:
STY: fix blank lines

Fix PEP8 compliance for blank lines after functions.

aburrell committed:
Merge branch 'rc_0_0_6' into no_remote_files

jklenzing committed:
Merge pull request #237 from pysat/cover

BUG: Revert integration to GitHub service for coveralls on MacOS

aburrell committed:
Merge pull request #240 from pysat/no_remote_files

Fix DMSP SSUSI file format change

jklenzing commented on pysatNASA/instruments/methods/cdaweb.py:
```diff
@@ -566,6 +590,52 @@ def download(date_array, data_path, tag='', inst_id='', supported_tags=None,
+    fname : str
+        Name of file on the remote server.
+    temp_path : str
+        Path to temporary directory. (Default=None)
```
```suggestion
        Path to temporary directory. Must be specified if zip_method is True.
        (Default=None)
```

jklenzing commented on pysatNASA/tests/test_instruments.py:
```diff
@@ -34,14 +42,20 @@
 # Create a new list of instruments with the option of forcing cdflib
 instruments['cdf'] = []
+
+# Create list of pandas instruments where this is not needed
```
```suggestion
# Create list of pandas instruments where this is not needed
# In general, this is for xarray instruments that are not supported
# by pysatCDF
```

jklenzing commented on pysatNASA/instruments/cnofs_vefi.py:
```diff
@@ -15,9 +23,10 @@
-Earth, https://geomag.us/models/pomme5.html) with the actual magnetometer
-measurements to help determine a set of calibration parameters for the gains,
-offsets, and non-orthogonality matrix of the sensor axes.  The calibrated
+Earth, https://geomag.colorado.edu/pomme-5-magnetic-model-of-the-earth.html)
```
```suggestion
Earth, https://geomag.org/models/pomme5.html)
```

jklenzing commented on pysatNASA/instruments/cnofs_vefi.py:
```diff
@@ -15,9 +23,10 @@
-Earth, https://geomag.us/models/pomme5.html) with the actual magnetometer
-measurements to help determine a set of calibration parameters for the gains,
-offsets, and non-orthogonality matrix of the sensor axes.  The calibrated
+Earth, https://geomag.colorado.edu/pomme-5-magnetic-model-of-the-earth.html)
+with the actual magnetometer measurements to help determine a set of calibration
+parameters for the gains, offsets, and non-orthogonality matrix of the sensor
+axes.  The calibrated
```
```suggestion
Earth, https://geomag.orgmodels/pomme5.html) with the actual magnetometer
measurements to help determine a set of calibration parameters for the gains,
offsets, and non-orthogonality matrix of the sensor axes.  The calibrated
```

jklenzing committed:
BUG: fix broken link

jklenzing commented on pysatNASA/instruments/cnofs_vefi.py:
```diff
@@ -15,7 +23,7 @@
 trajectory.  In order to provide an in-flight calibration of the magnetic field
 data, we compare the most recent POMME model (the POtsdam Magnetic Model of the
-Earth, https://geomag.us/models/pomme5.html) with the actual magnetometer
+Earth, https://geomag.orgmodels/pomme5.html) with the actual magnetometer
```
```suggestion
Earth, https://geomag.org/models/pomme5.html) with the actual magnetometer
```

jklenzing replied:
BUG: link

jklenzing committed:
BUG: link

aburrell commented on .github/workflows/main.yml:
```diff
@@ -71,7 +74,39 @@ jobs:
+          COVERALLS_PARALLEL: true
+        run: |
+          curl -sL https://coveralls.io/coveralls-linux.tar.gz | tar -xz
+          ./coveralls done --build-number ${{ github.run_number }}
```
```suggestion
          ./coveralls done --build-number ${{ github.run_number }}
 
```

aburrell commented on .zenodo.json:
```diff
@@ -1,12 +1,25 @@
+    "ionosphere",
+    "atmosphere",
+    "thermosphere",
+    "magnetosphere",
```
```suggestion
    "magnetosphere",
    "mars",
```

aburrell commented on pyproject.toml:
```diff
@@ -44,15 +44,16 @@ dependencies = [
 pysatcdf = ["pysatCDF"]
 test = [
-  "coveralls < 3.3",
+  "coveralls",
```
So, the coveralls folks don't recommend installing coveralls with the test.  If we remove this, the coveralls-actions *might* work.

jklenzing replied:
Fixed with #241

jklenzing committed:
STY: whitespace

jklenzing committed:
MAINT: use coveralls app

jklenzing committed:
BUG: whitespace

jklenzing committed:
MAINT: use coveralls app in rc test

jklenzing committed:
DOC: update changelog

jklenzing committed:
Update CHANGELOG.md

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>

jklenzing committed:
Merge pull request #241 from pysat/maint/coveralls

jklenzing commented on pysatNASA/instruments/methods/cdaweb.py:
```diff
@@ -777,7 +847,7 @@ def list_remote_files(tag='', inst_id='', start=None, stop=None,
             if 'month' in search_dir['keys']:
                 search_times = pds.date_range(start,
                                               stop + pds.DateOffset(months=1),
-                                              freq='M')
+                                              freq='ME')
```
M (month) was changed to ME (month end) with pandas 2.2.

jklenzing commented on pysatNASA/instruments/methods/cdaweb.py:
```diff
@@ -789,7 +859,7 @@ def list_remote_files(tag='', inst_id='', start=None, stop=None,
                 else:
                     search_times = pds.date_range(start, stop
                                                   + pds.DateOffset(years=1),
-                                                  freq='Y')
+                                                  freq='YE')
```
Similar change with pandas 2.2

rstoneback replied:
Thanks for adding a version check for the pandas stuff.

jklenzing reviewed (commented):
pandas 2.2 added new frequency offset aliases and deprecated the old ones. Had not realized the new syntax was this new. Tests will fail under older versions of pandas. Options
- Add a minimum pandas version
- Drop back to the old notation, add an issue and document in code, accept deprecation warning for now
- Some sort of hybrid where both are supported via try / except 🧟

jklenzing commented:
> pandas 2.2 added new frequency offset aliases and deprecated the old ones. Had not realized the new syntax was this new. Tests will fail under older versions of pandas. 

According to [SCEP0000](https://scientific-python.org/specs/spec-0000/), pandas 2.1 support should be maintained through Q3 of 2025. pandas 3.0 (unknown release date) will drop the deprecated 'M' and 'Y'. Leaning toward using an if else statement on the pandas version, linked to an issue describing what's up.

jklenzing committed:
MAINT: ensure backward compat with pandas

JonathonMSmith commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+download = functools.partial(cdw.download, supported_tags=download_tags)
+
+# Set the list_remote_files routine
+list_remote_files = functools.partial(cdw.list_remote_files,
```
We haven't really settled this, but I'll ask again. There's no reason why we have to use cdasws here, but we could. Do we have a feeling for whether we want to always use cdasws when possible or only use it when necessary?

jklenzing replied:
I would lean toward using cdasws where it applies, unless there is good reason not to.

jklenzing committed:
STY: cleanup testlines in GA ops

JonathonMSmith commented on pysatNASA/instruments/maven_insitu_kp.py:
```diff
@@ -0,0 +1,104 @@
+list_files = functools.partial(mm_gen.list_files,
+                               supported_tags=supported_tags)
+# Set the download routine
+basic_tag = {'remote_dir': ''.join(('/pub/data/maven/insitu/kp-4sec/',
+                                    'cdfs/{year:04d}/{month:02d}')),
+             'fname': fname}
+download_tags = {'': {'': basic_tag}}
+download = functools.partial(cdw.download, supported_tags=download_tags)
+
+# Set the list_remote_files routine
+list_remote_files = functools.partial(cdw.list_remote_files,
```
```suggestion
download_tags = {'': {'': 'MVN_INSITU_KP-4SEC'}}
download = functools.partial(cdw.cdas_download, supported_tags=download_tags)

# Set the list_remote_files routine
list_remote_files = functools.partial(cdw.cdas_list_remote_files,
```

JonathonMSmith commented on pysatNASA/instruments/maven_mag.py:
```diff
@@ -0,0 +1,101 @@
+list_files = functools.partial(mm_gen.list_files,
+                               supported_tags=supported_tags)
+# Set the download routine
+basic_tag = {'remote_dir': ''.join(('/pub/data/maven/mag/l2/sunstate-1sec',
+                                    '/cdfs/{year:04d}/{month:02d}')),
+             'fname': fname}
+download_tags = {'': {'': basic_tag}}
+download = functools.partial(cdw.download, supported_tags=download_tags)
+
+# Set the list_remote_files routine
+list_remote_files = functools.partial(cdw.list_remote_files,
+                                      supported_tags=download_tags)
```
```suggestion
download_tags = {'': {'': 'MVN_MAG_L2-SUNSTATE-1SEC'}}
download = functools.partial(cdw.cdas_download, supported_tags=download_tags)

# Set the list_remote_files routine
list_remote_files = functools.partial(cdw.cdas_list_remote_files,
                                      supported_tags=download_tags)
```

JonathonMSmith commented on pysatNASA/instruments/maven_sep.py:
```diff
@@ -0,0 +1,116 @@
+                               supported_tags=supported_tags)
+
+# Set the download routine
+basic_tag = {'remote_dir': ''.join(('/pub/data/maven/sep/l2/s1-cal-svy-full',
+                                    '/{year:04d}/{month:02d}')),
+             'fname': fname}
+
+basic_tag2 = {'remote_dir': ''.join(('/pub/data/maven/sep/l2/s2-cal-svy-full',
+                                    '/{year:04d}/{month:02d}')),
+              'fname': fname2}
+
+download_tags = {'s1': {'': basic_tag},
+                 's2': {'': basic_tag2}}
+
+# Set the download routine
+download = functools.partial(cdw.download, supported_tags=download_tags)
+
+# Set the list_remote_files routine
+list_remote_files = functools.partial(cdw.list_remote_files,
+                                      supported_tags=download_tags)
```
```suggestion
download_tags = {'s1': {'': 'MVN_SEP_L2_S1-CAL-SVY-FULL'},
                 's2': {'': 'MVN_SEP_L2_S2-CAL-SVY-FULL'}}

# Set the download routine
download = functools.partial(cdw.cdas_download, supported_tags=download_tags)

# Set the list_remote_files routine
list_remote_files = functools.partial(cdw.cdas_list_remote_files,
                                      supported_tags=download_tags)
```

jklenzing commented on docs/archival.rst:
```diff
@@ -0,0 +1,70 @@
+
+In this case, note that the pysat 'name' label is output to three different
+metadata values required by the ITSP standards. Additionally, the
+:py:attr:`export_pysat_info` option is set to false here. This drops several
+internal :py:mod:`pysat` metadata values before writing to file.
```
```suggestion
:py:attr:`export_pysat_info` option is set to false here. This drops several
internal :py:mod:`pysat` metadata values before writing to file. Note that
this includes the default acknowledgements and references objects. These
are set separately to avoid conflicts between the original dataset and the
new dataset, as well as keeping in line with requirements with potentially
different data servers.
```

jklenzing committed:
Apply suggestions from code review

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>
Co-authored-by: Jonathon Smith <36175570+JonathonMSmith@users.noreply.github.com>

jklenzing committed:
Merge pull request #243 from pysat/maint/pandas_freq

MAINT: ensure backward compat with pandas

jklenzing committed:
Apply suggestions from code review

jklenzing committed:
DOC: formosat note

jklenzing committed:
DOC: vefimag warning info

rstoneback commented on docs/archival.rst:
```diff
@@ -52,7 +52,11 @@ preferred formats.  An example of this is:
-internal :py:mod:`pysat` metadata values before writing to file.
+internal :py:mod:`pysat` metadata values before writing to file. Note that
+this includes the default acknowledgements and references objects. These
+are set manually to avoid conflicts between the original dataset and the
```
I checked out the REACH script but I don't see where an acknowledgement or references is set. Want to make sure users understand how to set information in this configuration.

jklenzing replied:
should be fixed with a reference to the point in the code

rstoneback commented on ACKNOWLEDGEMENTS.md:
```diff
@@ -0,0 +1,29 @@
+- NASA Scintillation Observations and Response of the Ionosphere to
+  Electrodynamics (SORTIE)
+- NASA Scintillation Prediction Observations Research Task (SPORT)
+
```
Certainly for CDAAC and added for packages that don't have it.

rstoneback reviewed (commented):
This review was for commits since my last review. After a brief intermission I'll be back to check out my previous comments as well :)

jklenzing committed:
Update docs/archival.rst

jklenzing committed:
Apply suggestions from code review

jklenzing committed:
BUG: error if temp_dir unset

jklenzing committed:
TST: test _get_file

jklenzing committed:
Update pysatNASA/instruments/methods/cdaweb.py

jklenzing committed:
Merge pull request #244 from pysat/bug/get_file

BUG/STY: update `_get_file` in download

jklenzing committed:
Update pysatNASA/instruments/templates/template_cdaweb_instrument.py

rstoneback commented on pysatNASA/instruments/templates/template_cdaweb_instrument.py:
```diff
@@ -11,7 +11,8 @@
 
-DO NOT include the NRL distribution statement in your new file.
+DO NOT include the NRL distribution statement in your new file. Statements for new
+files will have to be approved by NRL separately before the versioned release.
```
👎

rstoneback replied:
I'm checking with a lawyer on these NRL statements.

rstoneback replied:
NRL does not have authority over pysat code which also means NRL does not get to decide which code to approve or disapprove from public release. I'd like to discuss a path forward in the meeting today.

jklenzing committed:
STY: fix 80 char line length

jklenzing commented on pysatNASA/instruments/templates/template_cdaweb_instrument.py:
```diff
@@ -1,8 +1,20 @@
 
 Copy and modify this file as needed when adding a new Instrument to pysat.
 
+DO NOT include the NRL distribution statement in your new file. Statements for
+new files will have to be approved by NRL separately before the versioned
+release.
```
```suggestion
DO NOT include the NRL distribution statement in your new file. Contributions by
NRL developers will need to be cleared separately.
```

jklenzing replied:
@rstoneback, is this sufficient? It is difficult to suggest alternatives when there is no feedback.

rstoneback replied:
Apologies for the delay.
```suggestion
DO NOT include the NRL distribution statement in your new file. Contributions
by NRL developers will need to be cleared for classified or controlled information 
separately from the pysat pull request process.
```

jklenzing committed:
MAINT: cycle ops env test to 3.9

jklenzing committed:
DOC: update changelog

jklenzing committed:
BUG: apt get

jklenzing committed:
Merge pull request #246 from pysat/maint/ops_39

MAINT: cycle ops tests

jklenzing committed:
Update pysatNASA/instruments/templates/template_cdaweb_instrument.py

Co-authored-by: Russell Stoneback <github@stoneris.com>

jklenzing commented on pysatNASA/instruments/templates/template_cdaweb_instrument.py:
```diff
@@ -1,8 +1,20 @@
 Copy and modify this file as needed when adding a new Instrument to pysat.
 
+DO NOT include the NRL distribution statement in your new file. Contributions
+by NRL developers will need to be cleared for classified or controlled information 
+separately from the pysat pull request process.
```
```suggestion
by NRL developers will need to be cleared for classified or controlled 
information separately from the pysat pull request process.
```

jklenzing committed:
Apply suggestions from code review

Co-authored-by: Russell Stoneback <github@stoneris.com>

jklenzing commented on pysatNASA/instruments/templates/template_cdaweb_instrument.py:
```diff
@@ -1,8 +1,22 @@
 Copy and modify this file as needed when adding a new Instrument to pysat.
 
+DO NOT include the NRL distribution statement in your new file. Contributions
+by NRL developers will need to be cleared for classified or controlled 
```
```suggestion
by NRL developers will need to be cleared for classified or controlled
```

jklenzing committed:
Update pysatNASA/instruments/templates/template_cdaweb_instrument.py

jklenzing commented on CHANGELOG.md:
```diff
@@ -2,6 +2,51 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](https://semver.org/).
 
+## [0.0.6] - 2024-08-26
```
```suggestion
## [0.0.6] - 2024-10-04
```

jklenzing committed:
MAINT: update release, authors
